mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ChessBoardState.kt

```diff
@@ -0,0 +1,68 @@
+package ch.epfl.sdp.mobile.ui.game
+
+import androidx.compose.runtime.Stable
+
+/**
+ * An interface which represents the state of a [ChessBoard] composable. It display the current
+ * state of the game, and offers some ways for the [ChessBoard] to interact with it and indicate the
+ * user gestures.
+ *
+ * Each [ChessBoardState.Piece] has unique [Identifier], which is used to display some smooth
+ * animations if multiple pieces with the same ranks and colors exist.
+ *
+ * @param Identifier the type of the identifiers of each piece.
+ */
+@Stable
+interface ChessBoardState<Identifier> {
+
+  /** The different ranks which may be displayed by a [ChessBoard]. */
+  enum class Rank {
+    King,
+    Queen,
+    Rook,
+    Bishop,
+    Knight,
+    Pawn,
+  }
+
+  /** The different colors which may be displayed by a [ChessBoard]. */
+  enum class Color {
+    Black,
+    White,
+  }
```

Comment:
Why can't we use the enum that was already written [(this)](https://github.com/epfl-SDP/android/blob/main/mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/Rank.kt) ?

Reply:
The rationale was that the UI should not use the same types as the model, to separate concerns cleanly. In particular, the model enumeration may gain some behavior over time (to model moves for instance) which will be irrelevant to the UI layer. I think we do the same with profiles and people for example.

I agree that it looks a bit ridiculous right now. What would you say of revisiting this once we have a better idea of how the chess engine will be implemented, and how its `Rank` enumeration will look like ?